### PR TITLE
Added Hermite Distribution

### PR DESCRIPTION
**Pull request by Smit-create**
Added Hermite Distribution under Discrete Random Variables Types.
#### Release Notes

* stats
  * Added `HermiteDistribution`

### CODE REVIEW

File: sympy/stats/drv_types.py [hunk 1]
```diff
@@ -114,6 +116,91 @@ def Geometric(name, p):
...
+# Hermite distribution ---------------------------------------------------------
+
+
+class HermiteDistribution(SingleDiscreteDistribution):
+    _argnames = ('a1', 'a2')
+    set = S.Naturals0
+
+    @staticmethod
+    def check(a1, a2):
+        _value_check(a1.is_positive or a1.is_zero, 'Parameter a1 must be >= 0.')
```
Comment (czgdp1807) [inline sympy/stats/drv_types.py]: What about `a1.is_nonnegative`?

File: sympy/stats/drv_types.py [hunk 2]
```diff
@@ -114,6 +116,91 @@ def Geometric(name, p):
...
+        _value_check(a2.is_positive or a2.is_zero, 'Parameter a2 must be >= 0.')
```
Comment (czgdp1807) [inline sympy/stats/drv_types.py]: +1

File: sympy/stats/drv_types.py (hunk 1 above)
Comment (Smit-create) [reply sympy/stats/drv_types.py]: Yes, it works

File: sympy/stats/drv_types.py [hunk 3]
```diff
@@ -114,6 +116,91 @@ def Geometric(name, p):
...
+        a1, a2 = self.a1, self.a2
+        term1 = a1 * (exp(t) - 1)
+        term2 = a2 * (exp(2*t) - 1)
+        return exp(term1 + term2)
+
+    def _characteristic_function(self, t):
+        a1, a2 = self.a1, self.a2
+        term1 = a1 * (exp(I*t) - 1)
+        term2 = a2 * (exp(2*I*t) - 1)
+        return exp(term1 + term2)
```
Comment (czgdp1807) [inline sympy/stats/drv_types.py]: If we remove these two functions, the results are not that good looking as this, `Sum(num/den, (j, 0, k//2)).doit()` can not represented in closed form but we can remove this pair as IMHO, it's not `stats` moudle's concern to generate simplified as ideally it should have been handled by `Sum`.

Comment (Smit-create) [reply sympy/stats/drv_types.py]: I tried by removing them, but then, all the results i.e E, kurtosis, moment_generating_function, characteristic_function, skewness are not good as expected.
Removing them I get following results,
```
>>>E(X)
Sum(X*exp(-a1 - a2)*Sum(a1**(-2*_j + X)*a2**_j/(factorial(_j)*factorial(-2*_j + X)), (_j, 0, floor(X/2))), (X, 0, oo))

>>> moment_generating_function(X)(x)
Sum(exp(_x*x)*exp(-a1 - a2)*Sum(a1**(-2*_j + _x)*a2**_j/(factorial(_j)*factorial(-2*_j + _x)), (_j, 0, floor(_x/2))), (_x, 0, oo))

>>> characteristic_function(X)
Sum(exp(_x*I*x)*exp(-a1 - a2)*Sum(a1**(-2*_j + _x)*a2**_j/(factorial(_j)*factorial(-2*_j + _x)), (_j, 0, floor(_x/2))), (_x, 0, oo))

```
While kurtosis and skewness are taking too long to get computed.
So , in my opinion they should be kept added as of for now.

Comment (czgdp1807) [reply sympy/stats/drv_types.py]: Yeah they don't look good. It's okay to keep them though it's a bit ad-hoc.

**Comment by czgdp1807** [comment]
This PR appears to be ready. Will merge this if no objections are raised.

**Comment by czgdp1807** [comment]
@Smit-create Can you merge master in hermite locally then push the changes.
Follow,

1. git checkout master
2. git pull <remote-name-for-https://github.com/sympy/sympy/> master
3. git checkout hermite
4. git merge master
5. git push <remote-name-for-your-fork> hermite

**Comment by Smit-create** [comment]
> @Smit-create Can you merge master in hermite locally then push the changes.
> Follow,
> 
> 1. git checkout master
> 2. git pull remote-name-for-https://github.com/sympy/sympy/ master
> 3. git checkout hermite
> 4. git merge master
> 5. git push  hermite

Done

**Comment by Smit-create** [comment]
There came conflict now, so I resolved it.

**Comment by Smit-create** [comment]
@czgdp1807  Can you please review? Few hours before it passed all the checks and after it raised a conflict. After Resolving it and making a commit, failed a test in the file I haven't changed.

**Comment by czgdp1807** [comment]
LGTM. Let's wait if, someone has something to say anything. Will merge after that.

**Pull request merged**